### DISCUSSION TITLE
Docker on same host no longer working after upgrade to k3s v1.31.3

### DISCUSSION THREAD

**Discussion by viceice** [k3s-io/k3s#11487] [Q&A]
No sure if this is the right place to ask but i've a problem with my infra setup.

I've a test cluster with a controller and worker node.
On both nodes is the latest docker v27.4.1 installed.
I've running a ci worker container (github-runner) on the controller node inside docker.
That worker is responsible to deploy new k3s versions to my cluster (it's calling the k3s install script via ssh).

Since the upgrade to k3s v1.31.3+k3s1 (from v1.30.6+k3s1) the networking of docker is broken until a system restart.
This is causing the job inside the ci worker to timeout and fail.

I assume it's a new networking thing of kubernetes v1.31 which cleans docker related routes or firewall rules? 

Any help would be apreciate.
If no solution is found, then i'll need to move the docker hosts to separate vm's 😕

**Comment by brandond** [comment]
Are you using k3s with --docker (cri-dockerd) or with containerd?

It is never a good idea to run multiple container runtimes on one host, or to run unamanged containers alongside Kubernetes pods.

**Reply by viceice** [reply]
> Are you using k3s with --docker (cri-dockerd) or with containerd?

i'm using the default containerd backend.

> It is never a good idea to run multiple container runtimes on one host, or to run unamanged containers alongside Kubernetes pods.

I know, i probably move it to kubernetes too, but it needs to be able to self-update. so updating k3s from container 😏

**Reply by brandond** [reply]
Am I understanding properly that you're using a standalone Docker container to upgrade K3s on the host?

This is what the system upgrade controller is for, check the k3s docs.

**Reply by viceice** [reply]
yes, but that's not the issue. the issue is that there are network issues after k3s upgrade and I need to reboot the system to get it working again.

**Comment by viceice** [comment]
I've now also had connectivity issues inside another k3s cluster (no docker installed).
I've updated from v1.30.7+k3s1 to v1.30.8+k3s1.
my mosquitto server was no longer able to talk to anything via udp (mqtt).
A reboot solved the issue.

I'm running dualstack k3s with calico (tigera operator).
encapsulation is `VXLAN` with outgoing nat enabled.
